packages/router/src/lib/operations/prepare.ts

```diff
@@ -200,20 +206,78 @@ export const prepare = async (
   }
 
   logger.info("Validated input", requestContext, methodContext);
-  logger.info("Sending receiver prepare tx", requestContext, methodContext);
 
-  const receipt = await contractWriter.prepare(
-    invariantData.receivingChainId,
-    {
-      txData: invariantData,
-      amount: receiverAmount,
-      expiry: receiverExpiry,
-      bidSignature,
-      encodedBid,
+  if (isRouterContract) {
+    const routerRelayerFeeAsset = utils.getAddress(
+      config.chainConfig[invariantData.receivingChainId].routerContractRelayerAsset ?? AddressZero,
+    );
+    const relayerFeeAssetDecimal = await txService.getDecimalsForAsset(
+      invariantData.receivingChainId,
+      invariantData.receivingAssetId,
```

Comment:
wrong asset

Reply:
It's on router side, so it should be receiving assetId, right?